### PR TITLE
fix #589

### LINKED ISSUE #586: [BUG] chat isn't visible
### Description

When you connect to chat as anonymous and send messages, the other person doesn't receive messages and what you type isn't visible.

### Screenshots

Screenshot (212)

### Additional information

_No response_

### 🥦 Browser

Google Chrome

### 👀 Have you checked if this issue has been raised before?

- [X] I checked and didn't find similar issue

### 🏢 Have you read the Contributing Guidelines?

- [X] I have read and understood the rules in the [Contributing Guidelines](https://github.com/Dun-sin/Whisper/blob/main/CONTRIBUTING.md)

### Are you willing to work on this issue ?

Yes I am willing to submit a PR!

### LINKED ISSUE #589: [BUG] Volume slider in settings has a width of 0
### Description

The volume slider has a width of 0 so it's unusable.

### Screenshots

image

### Additional information

It can be fixed by setting the width of the `div.rs-form-control-wrapper` wrapping the slider to 100%. There isn't an option to pass a class to that div as far as I know, so it will require custom CSS.

### 🥦 Browser

Other

### 👀 Have you checked if this issue has been raised before?

- [X] I checked and didn't find similar issue

### 🏢 Have you read the Contributing Guidelines?

- [X] I have read and understood the rules in the [Contributing Guidelines](https://github.com/Dun-sin/Whisper/blob/main/CONTRIBUTING.md)

### Are you willing to work on this issue ?

Yes I am willing to submit a PR!

### PR DESCRIPTION
**Pull request by MJinH**
# Fixes Issue

**My PR closes #586**

# 👨‍💻 Changes proposed(What did you do ?)
fixed the error where the chat wasn't displaying, and the issue where messages weren't being delivered when the other user sent.
fixed name and type errors
# ✔️ Check List (Check all the applicable boxes)

- [X] My code follows the code style of this project.
- [X] This PR does not contain plagiarized content.
- [X] The title and description of the PR is clear and explains the approach.

## Note to reviewers

# 📷 Screenshots
chat_error

### CODE REVIEW

**Comment by Dun-sin** [comment]
> # PR Compliance Checks
> Thank you for your Pull Request! We have run several checks on this pull request in order to make sure it's suitable for merging into this project. The results are listed in the following section.
> 
> ## Conventional Commit PR Title
> In order to be considered for merging, the pull request title must match the specification in [conventional commits](https://www.conventionalcommits.org/en/v1.0.0/). You can edit the title in order for this check to pass. Most often, our PR titles are something like one of these:
> 
> * docs: correct typo in README
> * feat: implement dark mode"
> * fix: correct remove button behavior
> 
> Linting Errors
> 
> * Found type "null", must be one of "feat","fix","docs","style","refactor","perf","test","build","ci","chore","revert"
> * No subject found

@MJinH 
Please fix this error, if you don't know how, try this:
- Read the error carefully or read the contributing file
- Checkout past merged Pull requests to see how they were done for it to be merged

> Note: you can edit your PR message, this PR won't be merged if the error isn't gone and you have a week max to fix it before the PR gets closed without being merged

File: src/lib/chatSocket.ts [hunk 1]
```diff
@@ -24,7 +24,7 @@ export default function useChatUtils(socket: Socket | undefined) {
               return;
             }
 
-            resolve(sentMessage);
+            resolve(message);
```
Comment (Dun-sin) [inline src/lib/chatSocket.ts:27]: why the change here?

File: src/server/handlers.ts [hunk 2]
```diff
@@ -230,11 +230,11 @@ const messageCounts: {
   [id: string]: number;
 } = {};
 
-export const SendMessageHandler = (socket: Socket) => {
+export const SendMessageHandler = (io: Server, socket: Socket) => {
```
Comment (Dun-sin) [inline src/server/handlers.ts:233]: do we use the io?

**Comment by MJinH** [comment]
actually, let me close this PR and create a new one.

File: src/server/handlers.ts [hunk 3]
```diff
@@ -230,11 +230,11 @@ const messageCounts: {
...
   socket.on(
     constants.NEW_EVENT_SEND_MESSAGE,
     async (
-      { senderId, message, time, chatId, containsBadword, replyTo },
+      { senderId, message, time, room, containsBadword, replyTo },
```
Comment (Dun-sin) [inline src/server/handlers.ts:237]: leave it as chatId

File: src/server/models/MessageModel.ts [hunk 4]
```diff
@@ -59,7 +59,7 @@ const messageSchema = new Schema<MessageSchemaDocument>(
       default: false,
     },
     replyTo: {
-      type: Schema.Types.ObjectId,
+      type: String,
```
Comment (Dun-sin) [inline src/server/models/MessageModel.ts:62]: why the change here?

**Review by Dun-sin — CHANGES_REQUESTED** [review]

**Comment by Dun-sin** [comment]
> actually, let me close this PR and create a new one.

okay, but keep in mind my code review here, so you fix it on your new PR
